app/routes/signups.js

```diff
@@ -77,10 +77,13 @@ router.post('/', (req, res) => {
 
       return helpers.sendResponse(res, 200, scope.response_message);
     })
-    .catch(NotFoundError, err => helpers.sendResponse(res, 404, err.message))
-    .catch(UnprocessibleEntityError, err => helpers.sendResponse(res, 422, err.message))
-    .catch(err => helpers.sendResponse(res, 500, err.message));
+    .catch(err => {
+      switch (err.name) {
```

Comment:
There's a better way to do it. Import bluebird and catch them separately: http://bluebirdjs.com/docs/api/catch.html#filtered-catch

![image](https://cloud.githubusercontent.com/assets/672669/20436592/89b69a26-adb9-11e6-90cb-ee57b15c84ed.png)